target/linux/ipq40xx/image/generic.mk

```diff
@@ -780,6 +780,25 @@ define Device/qxwlan_e2600ac-c2
 endef
 TARGET_DEVICES += qxwlan_e2600ac-c2
 
+define Device/teltonika_rutx10
+	$(call Device/FitImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := Teltonika
+	DEVICE_MODEL := RUTX10
+	SOC := qcom-ipq4018
+	DEVICE_DTS_CONFIG := config@5
+	KERNEL_INSTALL := 1
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	FILESYSTEMS := squashfs
+	SUPPORTED_DEVICES := teltonika,rutx10 teltonika,rutx
+	IMAGE_METADATA := "device_code": [".*"], "hwver": [".*"], "batch": [".*"], "serial": [".*"]
```

Comment:
What's that supposed to do? I've never seen anything like it?

Reply:
This is needed for the stock firmware to accept the factory image. Stock firmware is QSDK based and uses a modified sysupgrade. It checks the metadata for these JSON nodes.

Reply:
Please don't hack the OpenWrt metadata for factory flashing. Add a proper factory image with its own recipe instead.

Reply:
I added an `append-rutx-metadata`  image command